### PR TITLE
feat: implement vc-widget component

### PR DESCRIPTION
**Pull request by goldenmaya**
### Jira-link:

https://virtocommerce.atlassian.net/browse/ST-5269
### Artifact URL:
https://vc3prerelease.blob.core.windows.net/packages/vc-theme-b2b-vue-1.45.0-pr-863-4eb7-4eb732bd.zip

### CODE REVIEW

**Review by NaMax66 — APPROVED** [review]

File: client-app/ui-kit/components/molecules/widget/vc-widget.vue [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+        </div>
+      </slot>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref, watchEffect } from "vue";
+
+export interface IProps {
```
Comment (Andrew-Orlov) [inline client-app/ui-kit/components/molecules/widget/vc-widget.vue:70]: Component shouldn't export its own props.

File: client-app/ui-kit/components/molecules/layout-with-right-sidebar/vc-layout-with-right-sidebar.vue [hunk 2]
```diff
@@ -26,10 +26,14 @@ defineProps({
...
+  @apply flex flex-col gap-5 px-6;
 
   @media (min-width: theme("screens.lg")) {
-    @apply flex-row items-start gap-x-6;
+    @apply flex-row items-start gap-x-6 px-0;
+  }
+
+  .vc-widget--size--lg {
+    @apply -mx-4.5 lg:mx-0;
   }
```
Comment (asvishnyakov) [inline client-app/ui-kit/components/molecules/layout-with-right-sidebar/vc-layout-with-right-sidebar.vue:37]: Why is it here and how it's related to this layout?

File: client-app/pages/account/dashboard.vue [hunk 3]
```diff
@@ -56,17 +58,13 @@
...
-        :title="$t('pages.account.dashboard.orders_status_card.title')"
-        class="h-52 lg:h-auto lg:w-1/2"
-        shadow
-      ></VcCard>
+      <VcWidget :title="$t('pages.account.dashboard.orders_status_card.title')" class="h-48 lg:w-0 lg:grow" />
     </div>
 
     <!-- Commented due to acceptance criteria, will be used in future
-    <VcCard title="Users" class="h-52 mx-5 md:mx-0" shadow></VcCard> -->
+    <VcWidget title="Users" class="h-52"></VcWidget> -->
```
Comment (asvishnyakov) [inline client-app/pages/account/dashboard.vue:67]: Simply remove such dead code, we always can restore it from git history

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

File: client-app/ui-kit/components/atoms/types.d.ts [hunk 4]
```diff
@@ -12,14 +12,16 @@ declare module "@vue/runtime-core" {
     VcAddressLine: typeof Components.VcAddressLine;
     VcBadge: typeof Components.VcBadge;
     VcBreadcrumbs: typeof Components.VcBreadcrumbs;
+    VcButton: typeof Components.VcButton;
+    /** @deprecated use {@link VcWidget} instead. */
```
Comment (asvishnyakov) [inline client-app/ui-kit/components/atoms/types.d.ts:16]: Please add runtime warning too: https://github.com/VirtoCommerce/vc-theme-b2b-vue/blob/9cf522d938e28f36092c485a3d157554ae647a71/client-app/ui-kit/components/organisms/sliding-actions/vc-sliding-actions.vue#L36

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

File: client-app/ui-kit/components/molecules/widget/vc-widget.vue [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+    &--lg {
+      $sizeLG: &;
+
+      --padding-x: 1.25rem;
+      --padding-y: 1rem;
+      --header-height: 3.875rem;
+
+      @apply divide-none;
+    }
+  }
```
[the remark this answers is not in the document]
Comment (goldenmaya) [reply client-app/ui-kit/components/molecules/widget/vc-widget.vue:143]: this is done in order not to produce the same styles separately for each size in the header, slot and footer

File: client-app/ui-kit/components/molecules/layout-with-right-sidebar/vc-layout-with-right-sidebar.vue (hunk 2 above)
Comment (goldenmaya) [reply client-app/ui-kit/components/molecules/layout-with-right-sidebar/vc-layout-with-right-sidebar.vue:37]: Good question, by design for a mobile view large widgets (sections) have different margins than small ones. Should we set these margins for each page separately? We need to think about this.

File: client-app/ui-kit/components/molecules/widget/vc-widget.vue (hunk 5 above)
[the remark this answers is not in the document]
Comment (goldenmaya) [reply client-app/ui-kit/components/molecules/widget/vc-widget.vue:143]: @asvishnyakov  We can not use tailwind classes here, it has to be used separately in &__header, &__slot and &__footer for each size.

**Review by Andrew-Orlov — APPROVED** [review]

**Review by asvishnyakov — APPROVED** [review]

**Pull request merged**